backtracking/minimax.cpp

```diff
@@ -1,27 +1,40 @@
+/**
+ * @file
+ * @brief returns which is the longest/shortest number
+ */
+#include <algorithm>
 #include <cmath>
 #include <iostream>
 #include <vector>
 
-using std::cout;
-using std::endl;
-using std::max;
-using std::min;
-using std::vector;
-
-int minimax(int depth, int node_index, bool is_max, vector<int> scores,
-            int height) {
-    if (depth == height)
+/**
+ * Check which number is the maximum/minimum in the array
+ * @param depth depth of array
+ * @param node_index current index in array
+ * @param is_max if current index is the longest number
+ * @param scores saved numbers in vector
+ * @return if maximum or not
+ */
+int minimax(int depth, int node_index, bool is_max,
+            std::vector<const int>*& scores, int height) {
```

Comment:
```suggestion
            const std::vector<int>& scores, int height) {
```
this is important correction